lib/Transforms/Utils/LoopSimplifyId.cpp

```diff
@@ -0,0 +1,24 @@
+//===- LoopSimplifyId.cpp - ID for the Loop Canonicalization Pass ----------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+
+#include "llvm/Pass.h"
+#include "llvm/Analysis/AliasAnalysis.h"
+#include "llvm/Analysis/LoopInfo.h"
+#include "llvm/Analysis/ScalarEvolution.h"
+#include "llvm/Transforms/Utils/LoopSimplify.h"
+
+using namespace llvm;
+
+char LoopSimplify::ID = 0;
+
+// Publicly exposed interface to pass...
+// This is in a separate file instead of LoopSimplify.cpp which brings in many dependencies
+// increasing the size of dxilconv.dll by (...TODO..) kB.
```

Comment:
I would expect to see a comment on the need for a file above the includes. I'm not sure if the specific reduction is necessary here, but the TODO suggests something is missing.

Reply:
I have removed the TODO. It is not easily identifiable what are the actual savings resulting from just this one change because the sum of the individual size savings of the three changes I did is higher than the actual number so there is some overlap.